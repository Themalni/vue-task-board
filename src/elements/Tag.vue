<template>
  <div :class="`tag text-sm tag-${type}`">
    {{ type }}
  </div>
</template>

<script>
export default {
  name: 'Tag',
  data: () => ({}),
  props: {
    type: {
      type: String
    }
  }
}
</script>

<style lang="scss">
.tag {
  padding: 0.05em 0.6em 0.3em;
  border-radius: 3px;
  border: 1px solid transparent;
}
.tag-feature {
  background-color: $olive;
  color: $white;
}
.tag-bugfix {
  background-color: $crimson;
  color: $white;
}
.tag-default {
  background-color: $white-smoke;
  border-color: darken($white-smoke, 5%);
}
</style>
