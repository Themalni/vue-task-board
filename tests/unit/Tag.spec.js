import { mount } from '@vue/test-utils'
import Tag from '@/elements/Tag'

let wrapper
let tag

beforeEach(() => {
  wrapper = mount(Tag)
  tag = wrapper.find('.tag')
})

describe('Button', () => {
  it('does contain `tag` element', () => {
    expect(tag.classes('tag')).toBe(true)
  })
  it('renters correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('does add class `tag-feature` when value is provided to prop `type`', () => {
    const propsData = {
      type: 'feature'
    }
    wrapper.setProps(propsData)
    expect(tag.classes('tag-feature')).toBeTruthy()
  })
})
