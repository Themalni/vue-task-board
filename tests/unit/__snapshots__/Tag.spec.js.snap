// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renters correctly 1`] = `
<div class="tag text-sm tag-undefined">

</div>
`;
